traceapp/tmpl/doc.go

```diff
@@ -3,4 +3,4 @@
 // This package needs to be go generated after making changes to template files.
 package tmpl
 
-//go:generate go run -tags=dev data_generate.go
+//go:generate vfsgen -tags=dev -source="sourcegraph.com/sourcegraph/appdash/traceapp/tmpl".Data -output-tags=!dev
```

Comment:
> Is that correct?

Yes, that's correct. But let me elaborate on "-tags=dev is needed because traceapp/tmpl/data.go has // +build dev" point.

vfsgen works with a Go value as input. We access it by specifying two things in -source, the import path, and the name of an exported variable defined in top level package scope.

Counterintuitively, those 2 parameters are not enough to specify a Go value precisely. For example, consider a package with two Go files:

```Go
// +build foo

package a

var SomeVariable = "this is the value of SomeVariable in this package"
```


```Go
// +build bar

package a

var SomeVariable = "this is a completely different value! but same variable!"
```

As you can see, the value of "import/path/of/package/a".SomeVariable is meaningless unless you specify the build environment, which includes builds tags.

I used the familiar -tags flag to control build tags for source. The -source syntax is inspired by the `gorename` tool, but since that tool doesn't deal with values of variables but rather just their names, it doesn't ask the user to specify build tags.

Perhaps there's some way of combining the 3 parameters that are needed to specify the source (import path, variable name, build tags) into a single parameter to make it more clear? I couldn't think of one so far, but that's an idea to consider.

Okay, back to other flags. In the example above, the only other flag specified is -output-tags. But it has nothing to do with the source -tags, it controls the tags that are written in the output file. Unlike other output parameters, this one is harder to have a sensible default value for. But if we were to specify all output parameters explicitly, the command above may look like this:

```
vfsgen -tags=dev -source="sourcegraph.com/sourcegraph/appdash/traceapp/tmpl".Data -output-filename="fancy_filename.go" -output-package="fancy_package_name" -output-variable="FancyVariable" -output-tags="linux !amd64 fancy_custom_tag".
```

However, it seemed possible to deduce good default values for the filename, package name, variable name for the generated output. But not for output build tag constraints. In this particular case, the output tags is the boolean inverse of the input tags, but what if input tags were like "linux dev" or something?